Anytype/Sources/PresentationLayer/TextEditor/Set/Models/SetDocument.swift

```diff
@@ -29,15 +29,14 @@ class SetDocument: SetDocumentProtocol {
     
     var forPreview: Bool { document.forPreview }
     
-    var dataviews: [BlockDataview] {
-        return document.children.compactMap { info -> BlockDataview? in
-            if case .dataView(let data) = info.content {
-                if let blockId = inlineParameters?.blockId {
-                    return info.id == blockId ? data : nil
-                } else {
-                    return data
-                }
-            }
+    var blockDataview: BlockDataview? {
+        let blockId = inlineParameters?.blockId ?? SetConstants.dataviewBlockId
+        guard let blockInfo = document.infoContainer.get(id: blockId) else {
+            return nil
+        }
+        if case .dataView(let data) = blockInfo.content {
+           return data
+        } else {
```

Comment:
`document.children` are updated on middle update, but activeViewId has offline logic in Sets, that is why we should get block from the `infoContainer`